### PR TITLE
🐛Allow amp-analytics to layout correctly in FIE

### LINKED ISSUE #19678: Amp-analytics doesn't get laid out correctly in FIE
The issue is affecting all <amp-analytics> when placed within the AMPHTML ad creative. 

The issue is related to <amp-analytics> being a position fixed element, and the runtime fails to measure its correct `layoutRect` if it is placed within an iframe. This is more likely to happen if the ad slot is placed three viewport away from top. In such cases, the <amp-analytics> element will only be laid out when runtime is idle, which introduces delay to measurement.

Another issue, is that <amp-analytics> gets `display: inline-block` `position:fixed`, but not `left: 0`, so it is sometimes placed outside of viewport x-axis. Which leads to runtime not loading the element. 
https://github.com/ampproject/amphtml/blob/781be6ba70ca08e3ce8095ec9041b6e0dd24e4e7/src/service/resource.js#L727-L732

### PR DESCRIPTION
**Pull request by zhouyx**
Closes #19678 to @jridgewell @lannka 

cc @keithwrightbos 
cc @calebcordry on amp-story-auto-ads change.

### CODE REVIEW

File: examples/analytics-iframe-transport-remote-frame.html [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <meta charset="UTF-8">
     <title>Requests Frame</title>
     <script>
+      console.log('Created Iframe!!!!!!');
```
Comment (keithwrightbos) [inline examples/analytics-iframe-transport-remote-frame.html]: remove

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 2]
```diff
@@ -114,6 +115,9 @@ export class AmpAnalytics extends AMP.BaseElement {
 
   /** @override */
   isAlwaysFixed() {
+    if (isInFie(this.element)) {
```
Comment (keithwrightbos) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Why not just return !isInFie(this.element);?

File: examples/analytics-iframe-transport-remote-frame.html (hunk 1 above)
Comment (zhouyx) [reply examples/analytics-iframe-transport-remote-frame.html]: I added this intentionally to inform that the iframe has loaded w/o looking at network request : ) 

Anyhow I think I could change my example page to not use transport: iframe. I'll remove that

**Comment by zhouyx** [comment]
@keithwrightbos Yes! I did go through the element list that's supported within AMP Ad. And amp-analytics is the only component with position fixed right now. 

Eventually I think FIE should handle the layout of its resources automatically. I am hoping that to come with the new layer change : ) @jridgewell

**Comment by zhouyx** [comment]
There is also amp-lightbox that's position fixed, but its layout is handled differently.

**Comment by keithwrightbos** [comment]
Can we add a runtime check that will report if any amp elements are position fixed within FIE? Want to make sure this doesn't happen again and of course it can be removed once layers is rolled.  I'll approve so you can address in a different PR.

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.css [hunk 3]
```diff
@@ -20,6 +20,20 @@
...
+  /* Fixed to make position independent of page other elements. */
+  position: fixed !important;
+  top: 0 !important;
+  left: 0 !important;
+  width: 1px !important;
+  height: 1px !important;
+  overflow: hidden !important;
+  visibility: hidden;
+}
+
```
Comment (calebcordry) [inline extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.css]: LGTM. Thanks!

Comment (lannka) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.css]: @calebcordry could you help to patch the change locally and make sure it works for amp-story-auto-ads (i think it does, just to be super cautious)

Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.css]: I checked a few local test files, they look good.

**Pull request merged**